api/v1beta1/azuremachine_types.go

```diff
@@ -118,6 +118,8 @@ type AzureMachineSpec struct {
 	// SubnetName selects the Subnet where the VM will be placed
 	// +optional
 	SubnetName string `json:"subnetName,omitempty"`
+
+	NetworkInterfaces []AzureNetworkInterface `json:"networkInterfaces,omitempty"`
```

Comment:
Can we add a doc comment for this field? And if it's optional we can also add `// +optional` as well for the kubebuilder generation.